File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        
        def dfs(root, p, q):
            if not root:
                return None 
            min_val = min(p.val, q.val)
            max_val = max(p.val, q.val)
            if min_val <= root.val <= max_val:
                return root
            
            if root.val < min_val:
                return dfs(root.right, p, q)
            else:
                return dfs(root.left, p, q)
        return dfs(root, p, q)
        

File: Data Structures & Algorithms/max-consecutive-ones-ii/submission-3.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max_ones = 0;
        int curr_ones = 0;
        int left = 0;
        int flipped = 0;

        for (int right = 0; right < nums.length; right++){
            if (nums[right] == 1){
                curr_ones ++;
            }

            if (nums[right] == 0){
                flipped ++;
                curr_ones ++;
            }

            while (flipped > 1){
                if (nums[left] == 0){
                    flipped -=1;
                }
                curr_ones -= 1;
                left += 1;
            }
            max_ones = Math.max(max_ones, curr_ones);
        }
        return max_ones;
    }
}
